.github/actions/build-push-image/action.yaml

```diff
@@ -88,34 +82,65 @@ runs:
       shell: bash
       run: |
         go mod download
-    - name: Build and Push Operator to Docker Registry
+    - name: Build all platforms
+      shell: bash
+      run: make all-platforms
+    - name: Fail if send input flags are both unset
+      if: ${{ inputs.push_to_docker != 'true' && inputs.push_to_quay != 'true' }}
+      shell: bash
+      run: exit 1
+    - name: Compute main build tag
+      id: tag
+      shell: bash
+      run: |
+        main_tag=${{ inputs.push_to_docker == 'true' && format('{0}:{1}', inputs.repository, inputs.version) || format('quay.io/{0}:{1}', inputs.repository, inputs.version)}}
+        echo "main_tag=$main_tag" >> $GITHUB_OUTPUT
+    - name: Login to docker registry
       if: ${{ inputs.push_to_docker == 'true' }}
-      uses: docker/build-push-action@v3
+      uses: docker/login-action@v2
       with:
-        context: .
-        file: ${{ inputs.file }}
-        build-args: VERSION=${{ inputs.version }}
-        platforms: ${{ inputs.platforms }}
-        tags: ${{ inputs.repository }}:${{ inputs.version }}
-        cache-from: type=gha
-        cache-to: type=gha,mode=max
-        push: true
+        username: ${{ inputs.docker_username }}
+        password: ${{ inputs.docker_password }}
     - name: Login to quay.io registry
       if: ${{ inputs.push_to_quay == 'true' }}
       uses: docker/login-action@v2
       with:
         registry: quay.io
         username: ${{ inputs.quay_username }}
         password: ${{ inputs.quay_password }}
-    - name: Build and Push Operator to Quay Registry
-      if: ${{ inputs.push_to_quay == 'true' }}
+    - name: Build and Push Operator to ${{ steps.tag.outputs.main_tag }}
       uses: docker/build-push-action@v3
       with:
         context: .
         file: ${{ inputs.file }}
         build-args: VERSION=${{ inputs.version }}
         platforms: ${{ inputs.platforms }}
-        tags: quay.io/${{ inputs.repository }}:${{ inputs.version }}
         cache-from: type=gha
         cache-to: type=gha,mode=max
+        sbom: true
         push: true
+        tags: ${{ steps.tag.outputs.main_tag }}
+    - name: Push To Quay
+      if: ${{ inputs.push_to_quay == 'true' && inputs.push_to_docker == 'true' }}
```

Comment:
Why does push to Quay depend on push to docker?

Reply:
We have both input flags to push to docker AND / OR quay. In top of that we know that build and push must push to at lease some repo, but depending on the inputs, we do not know which one it will be. The table of truth looks like this:

- Both flags false => **Fail the job!**
- Push to docker / NO quay => build and push will push to docker and this must be skipped 
- Push to quay / NO docker => build and push will push to quay, so this must be skipped also
- Both Push docker & quay => build and push will push to docker, and this must push to quay now